Release/segundo parcial

carolosano opened the pull request.

Commit:
prueba para ver si hago el push

Commit:
segundo push que hacemos con el profe

carolosano committed:
Primer avance introduccion a model box

carolosano committed:
Segunda correccion de la segunda entrega

carolosano committed:
Resuelto conflicto con .DS_Store

carolosano committed:
Implementación de cambios adicionales para la tercera entrega

carolosano committed:
cambios solicitados para hacer Entrega Act N4

carolosano committed:
2da vuelta de cambios para la rama f tercera entrega

carolosano committed:
ver si el gitignore funciona

carolosano committed:
Elimina node_modules del seguimiento de Git

carolosano committed:
cambios que me pidio el profe

carolosano committed:
ya estarian todos los cambios agregados

carolosano committed:
agregue Producto.js + packAlternativo.js

carolosano committed:
ultimos cambios pendientes resueltos jueves 21.10

carolosano committed:
Cambie las instancias exportadas que no corresponden

carolosano committed:
entrega unidad4, saque los promts y anadi el dom y local storage

carolosano committed:
agregue un carrito, un array con el pack que se armo y despues para finalizar la com
pra

carolosano committed:
agregue los sweet alert

carolosano committed:
uso packAlternativo, saque las exportaciones de armarpack y finalizarCompra

carolosano committed:
cambios eliminacion del window, saque el evento incrustado en el html y lo puse en el js

carolosano committed:
saco una parte de los eventos

carolosano committed:
saque los comments

carolosano committed:
primeros cambios con api

carolosano committed:
aguege el readme

carolosano committed:
me olvide de anadir el readme

carolosano committed:
a ver si aparece

MVelasquez98 commented on CHANGELOG.md:
Esto esta incompleto, no solo es una plantilla, tenes que rellenarlo, con todo lo que hiciste en el proyecto, Actividades Obligatorias 1, 2, 3 y 4. Primer y Segundo Parcial, incluir fechas, linkear las pull request.

MVelasquez98 commented on script.js:
```diff
@@ -0,0 +1,235 @@
+
+// Crear nuevos datos en la API
+createDataButton.addEventListener('click', () => {
+  const newUser = {
```
¿Y esto que tiene que ver con el pryecto?

MVelasquez98 commented on script.js:
```diff
@@ -0,0 +1,235 @@
+
+// Función para mostrar datos en pantalla
+function displayData(data) {
+  outputDiv.innerHTML = `<pre>${JSON.stringify(data, null, 2)}</pre>`;
```
No es la idea del uso de una api, estas literalmente imprimiendo el JSON en el HTML:
![image](https://github.com/user-attachments/assets/93393cec-686d-476f-8761-b20c554689f6)
Tenes que trabajar con esta información y el uso de  DOM, además que la información debe ser relevante para el proyecto ¿de que sirve en un sitio de "packs" imprimir una lista de usuarios?

MVelasquez98 commented on script.js:
```diff
@@ -0,0 +1,235 @@
+const outputDiv = document.getElementById('output');
+
+// Endpoint base de la API
+const apiBaseUrl = 'https://6748dd015801f515359267a4.mockapi.io/c1/productos';
```
Creo que no tiene mucho sentido el mock que escogiste, ¿porque no tener los packs en la API y renderizarlos en el listado a escoger, en vez de que esten el html estatico?

MVelasquez98 commented on script.js:
```diff
@@ -0,0 +1,235 @@
+    document.getElementById("contact-form").reset();
+});
+
+  const packGrafico = new Producto(
```
Estos packs son los que tienen que estar en un JSON aparte y consumir mediante fetch. 
`api/packs.json`

MVelasquez98 commented on packAlternativo.js:
Si esto es una clase, el nombre del archivo es PackAlternativo.js, usamos siempre PascalCase para el nombre de las Clases

MVelasquez98 commented on script.js:
```diff
@@ -0,0 +1,235 @@
+import { Producto } from "./Producto.js";
+import PackAlternativo from "./packAlternativo.js";
```
Si esto es una clase, el nombre del archivo es PackAlternativo.js, usamos siempre PascalCase para el nombre de las Clases

MVelasquez98 commented on script.js:
```diff
@@ -0,0 +1,235 @@
+import { Producto } from "./Producto.js";
```
Las `{ }` son innecesarias

MVelasquez98 commented on Producto.js:
```diff
@@ -0,0 +1,16 @@
+}
+
+// Exportamos solo la clase Producto
+export { Producto };
```
Las `{ }` son innecesarias

MVelasquez98 commented on script.js:
```diff
@@ -0,0 +1,235 @@
+document.getElementById("contact-form").addEventListener("submit", function(event) {
+    // prevenir que el formulario realice la acción por defecto (refrescar la página x ej)
+    event.preventDefault();
+
```
Enciérralo en un bloque try-catch para capturar cualquier error inesperado y realizar una acción adecuada, como registrar el error o mostrar un mensaje de advertencia al usuario.
```suggestion
try {
```

MVelasquez98 commented on script.js:
```diff
@@ -0,0 +1,235 @@
+
+    // reseteo el formulario
+    document.getElementById("contact-form").reset();
+});
```
```suggestion
catch (error) {
     console.error('Error en el envío del formulario:', error);
  }
});
```

MVelasquez98 commented on script.js:
```diff
@@ -0,0 +1,235 @@
+getDataButton.addEventListener('click', () => {
+  fetch(apiBaseUrl)
+    .then(response => {
+      if (!response.ok) throw new Error('Error al obtener los datos');
```
Sumar un
`   console.error('Error:', error);
`

MVelasquez98 reviewed (changes requested):
**PUNTAJE POR CONSIGNA**:
La evaluación se realiza sobre una escala de 10 puntos, donde se aprueba con un mínimo de 4 puntos.

- **Asincronía y Manejo de Datos (3 puntos)**
⚠️ 0.5 puntos: Utiliza AJAX y JSON para obtener datos de forma asíncrona, pero no manejando adecuadamente promises y fetch para consumir alguna API mock .
❌ 0 puntos: La información estática no esta en formato JSON, no se se carga y utiliza de manera asíncrona. Los packs estan estaticos en el código, deberian estar en JSON.
❌ 0 puntos: Recorrido de colecciones siendo óptimo utilizando funciones de orden superior. No se renderiza correctamente lo obtenido de la api, se imprime en el HTML sin usar DOM.

- **Gestión del Proyecto y Buenas Prácticas con GitHub (7 puntos)**:
✅ 1 punto: Realización de commits necesarios sobre la rama de desarrollo utilizando Git.
✅ 1 punto: Creación de una rama release/segundo-parcial a partir de la rama develop utilizando Git.
✅ 1 punto: Correcta realización de la solicitud de revisión y aprobación de la pull request para fusionar la rama feature/cuarta-entrega con develop en GitHub.
✅ 1 punto: Proporcionar la URL de la pull request y la URL generada por GitHub Pages para la rama release/segundo-parcial como parte de la entrega.
❌ 0 puntos: No Implementa un archivo [CHANGELOG.md](https://github.com/Laboratoria/changelog-guide) documentando los cambios realizados en el proyecto en cada entrega o versión, siguiendo un formato estructurado.
❌ 0 puntos: No Configura [reglas en GitHub](https://docs.github.com/es/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/managing-a-branch-protection-rule) para bloquear push directos a las ramas develop y master, exigiendo pull requests con al menos un revisor aprobado para realizar merges. Incluir en el comentario de la pull request una captura con el estado de las reglas.
❌ 0 puntos: No Implementa una [plantilla (template) de pull request](https://docs.github.com/es/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository) en el repositorio para estandarizar las solicitudes de revisión y asegurarse de incluir in[formación relevante](https://axolo.co/blog/p/part-3-github-pull-request-template) (descripción, cambios realizados, pruebas realizadas, etc.).

**Nota**: 4(Cuatro)

**Comentarios**: En primer lugar, quiero reconocer el esfuerzo que pusiste en el proyecto. Sin embargo, hay varias áreas que necesitan atención para mejorar la cohesión y funcionalidad de tu trabajo, y poder aprobar el final de la materia.

- **Enfoque en el Simulador:** Parece que algunas características están un poco desconectadas del objetivo principal. Por ejemplo, imprimir una lista de usuarios no parece tener una relación lógica con la funcionalidad de los packs. Es importante que tu implementación sea coherente y que todas las piezas trabajen juntas en función del objetivo del simulador. Recomendación: Define claramente el propósito del simulador y revisa el diseño para asegurar que todos los componentes interactúen de manera significativa.

- **Asincronía y Manejo de Datos:** Aunque comenzaste a implementar AJAX y JSON, sería beneficioso abordar cómo se estructuran los datos. Considera mover los packs a un archivo JSON y cargarlos de manera asíncrona. Esto no solo mejora la organización, sino que también te ayuda a ganar puntos en la evaluación. Además, usa funciones de orden superior para recorrer y manipular colecciones de datos.

- **Buenas Prácticas en GitHub:** Existen áreas claras de mejora aca. Asegúrate de configurar correctamente tu flujo de trabajo en GitHub:
  - Implementa un `CHANGELOG.md` para llevar un registro de los cambios realizados en cada versión.
  - Configura reglas en las ramas principales para evitar pushes directos y exigir revisiones aprobadas. (tenes que incluir una captura, como pide la consigna)
  - Crea una plantilla para pull requests que facilite la revisión y la documentación de cambios efectuados.

- **Preparación para el Final:** Considera que este proyecto debe estar listo para ser defendido en el examen final. Intenta trabajar más en la cohesión y propósito del proyecto, asegurándote de que cada componente contribuya positivamente al simulador total.

Con estos ajustes y un enfoque claro, tu proyecto puede mejorar significativamente. ¡Ánimo y seguí adelante, que estás en el camino correcto para obtener un resultado más sólido!

carolosano commented:
@MVelasquez98 perfecto profe voy a ir haciendo estas correcciones, hasta cuando tengo tiempo de seguir perfeccionando el proyecto?

carolosano committed:
Agregue el json para que lo consuma la API y que no sea un contenido estatico en el HTML

carolosano committed:
Cambios ## [2.2.0] - 2025-07-24

- Se creó un archivo JSON para que la API consuma mis datos de los packs y los renderice, lo utilice de forma asincrona
- Implemento un recorrido óptimo de las colecciones utilizando funciones como map.

### 🔧 Cambios
- Se reemplazó el contenido estático de los packs en HTML por una carga dinámica desde el archivo `packs.json`, usando `fetch` y renderizado automático en el DOM.
- Corrección del error de lectura de `undefined` en `agregarPackAlCarrito`.
- Archivo `packs.json` para almacenar los datos de los packs de forma estructurada en formato JSON, con el objetivo de que la API lo consuma dinámicamente.
- Template de pull request (`.github/pull_request_template.md`) para estandarizar la revisión de cambios.
- Reglas de protección en ramas `main` y `develop` para forzar uso de Pull Requests con revisión.
- Estructura modular con `Producto.js` y `packAlternativo.js`.
- Renderizado básico de packs en el DOM.
- Funcionalidad para agregar packs al carrito y almacenarlos en `localStorage`.

carolosano commented:
@MVelasquez98 Hola profe, como le va tanto tiempo? Acá corregí lois cambios del segundo parcial que me quedaron pendientes, le pongo acá lo que me había corregido la ultima vez y fuí corrigiendo todos los puntos a continuaciòn:

⚠️ 0.5 puntos: Utiliza AJAX y JSON para obtener datos de forma asíncrona, pero no manejando adecuadamente promises y fetch para consumir alguna API mock .
❌ 0 puntos: La información estática no esta en formato JSON, no se se carga y utiliza de manera asíncrona. Los packs estan estaticos en el código, deberian estar en JSON.
❌ 0 puntos: Recorrido de colecciones siendo óptimo utilizando funciones de orden superior. No se renderiza correctamente lo obtenido de la api, se imprime en el HTML sin usar DOM.
❌ 0 puntos: No Implementa un archivo [CHANGELOG.md](https://github.com/Laboratoria/changelog-guide) documentando los cambios realizados en el proyecto en cada entrega o versión, siguiendo un formato estructurado.
❌ 0 puntos: No Configura [reglas en GitHub](https://docs.github.com/es/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/managing-a-branch-protection-rule) para bloquear push directos a las ramas develop y master, exigiendo pull requests con al menos un revisor aprobado para realizar merges. Incluir en el comentario de la pull request una captura con el estado de las reglas.
❌ 0 puntos: No Implementa una [plantilla (template) de pull request](https://docs.github.com/es/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository) en el repositorio para estandarizar las solicitudes de revisión y asegurarse de incluir in[formación relevante](https://axolo.co/blog/p/part-3-github-pull-request-template) (descripción, cambios realizados, pruebas realizadas, etc.).

MVelasquez98 commented on pull_request_template.md:
¿Por qué esta vacio? Tiene que ser justamente un template con todo lo necesario. Revisa la documentación, en que carpeta y demás. [doc github](https://docs.github.com/es/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository)
Si esta bien configurado, ya deberías poder usar esta plantilla en esta misma pull request.

carolosano replied:
<img width="1917" height="348" alt="Captura de pantalla 2025-07-24 a las 6 20 00 p  m" src="https://github.com/user-attachments/assets/1f81d880-b03f-46db-9458-9e39b0e54926" />
ahi vi la documentacion, se me hace un poco confuso, perdon profe, asi seria? la plantilla

MVelasquez98 commented:
Las reglas de github siguen mal configuradas.
Como se ve en tu repositorio:
<img width="948" height="398" alt="image" src="https://github.com/user-attachments/assets/7dac7b03-7011-4470-9212-d2927afed875" />
Como se tendría que ver:
<img width="924" height="242" alt="image" src="https://github.com/user-attachments/assets/09c13201-6c7e-49c1-ad6a-ea8e0989d6e0" />
El [video](https://youtu.be/qe6mPZXDNRM) explicativo del campus

MVelasquez98 commented on CHANGELOG.md:
```diff
@@ -0,0 +1,106 @@
+
+
+
+## [2.2.0] - 2025-07-24
```
No usamos numeros de versiones en ningun momento, en el changelog van todas las actividades obligatorias y parciales, enlazados correctamente en markdown la pr correspondiente

MVelasquez98 reviewed (changes requested):
<img width="395" height="579" alt="image" src="https://github.com/user-attachments/assets/9496ceaa-2921-4ae3-b413-5d7db5ab2a2c" />
Es bastante confuso para un carrito, hay productos repetidos, nunca se entiende de que pack vienen ¿se puede mejorar la experiencia de usuario? Diferenciando los paquetes en el carrito y tambien si estan repetidos agrupar con cantidad. Por otro lado, si agrego un pack personalizado se sobreescribe el carrito ¿es un comportamiento deseado?

carolosano commented:
> Las reglas de github siguen mal configuradas. Como se ve en tu repositorio: <img alt="image" width="948" height="398" src="https://private-user-images.githubusercontent.com/43156596/470525788-7dac7b03-7011-4470-9212-d2927afed875.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTMzOTIxODMsIm5iZiI6MTc1MzM5MTg4MywicGF0aCI6Ii80MzE1NjU5Ni80NzA1MjU3ODgtN2RhYzdiMDMtNzAxMS00NDcwLTkyMTItZDI5MjdhZmVkODc1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzI0VDIxMTgwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZjNTE4ZGZlYTI4NjlmMjVlZWIyZDU2ZjM5YjAwNWE0NGJiZjA2MjU5MjY4NDA3ODEwOWNmZDNhZGNmYzNmNmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Yg4_86Moh8L_G2uNOm2lq30Mj4Y6JciJ4ytne3LTXzg"> Como se tendría que ver: <img alt="image" width="924" height="242" src="https://private-user-images.githubusercontent.com/43156596/470526113-09c13201-6c7e-49c1-ad6a-ea8e0989d6e0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTMzOTIxODMsIm5iZiI6MTc1MzM5MTg4MywicGF0aCI6Ii80MzE1NjU5Ni80NzA1MjYxMTMtMDljMTMyMDEtNmM3ZS00OWMxLWFkNmEtZWE4ZTA5ODlkNmUwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzI0VDIxMTgwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc1NTg4M2E1Njk5NDUwYTlmMjhkNTViMjZkOGI3N2FkNmYyMGNhYWE4OTU3MDcxMTA3ZTdkMTFlOTYxYTA1NjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.JovhSWPjRIHV7YlX9PZWhHNl9syQXM8GglzzKwde8J4"> El [video](https://youtu.be/qe6mPZXDNRM) explicativo del campus

Okei perfecto, ahi lo hice para los Release/**

carolosano committed:
Create pull_request_template.md

carolosano committed:
Update pull_request_template.md

MVelasquez98 commented:
Las reglas siguen mal configuradas, la protección es para la rama máster, que es hacia donde estás haciendo la pull request, revisa la pr, mientras te deje hacer merge en verde, sigue mal configurada. Tiene que estar bloqueada en rojo esperando mi aprobación, como te mostré en la imagen

carolosano committed:
cambios requeridos del segundo parcial

corregi el changelog y el template de pull request

carolosano commented:
> Las reglas siguen mal configuradas, la protección es para la rama máster, que es hacia donde estás haciendo la pull request, revisa la pr, mientras te deje hacer merge en verde, sigue mal configurada. Tiene que estar bloqueada en rojo esperando mi aprobación, como te mostré en la imagen

@MVelasquez98 Ah perfectop, ahi lo configure para que sea a la rama master las reglas. Camie el archivo CHANGELOG.md y puse el template en el pull_request_template. Me podría profe decir si es que lo hice bien o sigue algo mal? vi su video explicativo, fue de mucha ayuda, lo que nose si entendí bien fue la documentacion para el template, pero creo q lo pude hacer. Ahora sigo con los cambios en el carrito que me pidió

carolosano committed:
Cambios en el UX del carrito

Puse para que solo se vea el nombre del pack elegido, que no se pueda poner mas de una vez el pack y que aparezca abajo el boton un "ya ha sido agregado este pack al carrito" para que el cliente no compre dos veces el mismo pack porque no tiene sentido. Tambien cambié el titulo de Carrito para que se diferencie mas

carolosano committed:
Agregue el ANEXO por si no llegaba a hacerlo para cuando me apruebes los cambios del  segundo parcial

MVelasquez98 reviewed (dismissed):
LGTM

carolosano commented:
@MVelasquez98 nose que hice que ahora no puedo hacer merge profe :(. mE podria ayudar?

MVelasquez98 commented:
La reléase está aprobada solo tenes que hacer merge, el anexo no entra en esta reléase igual

carolosano commented:
> La reléase está aprobada solo tenes que hacer merge, el anexo no entra en esta reléase igual

Me aparece esto de que hay conflictos
<img width="1077" height="507" alt="Captura de pantalla 2025-07-24 a las 10 36 21 p  m" src="https://github.com/user-attachments/assets/8e69a8b4-9c2c-4f2f-bacf-793ae825708e" />

MVelasquez98 commented:
Eso es porque agregaste el anexo antes, tenes conflictos de merge, tenes que resolverlos en tu local y volver a hacer push. Baja con pull lo último de develop y después hace rebase and merge de la rama de la reléase con develop, ahí te van aparecer todos los conflictos y tenes para aceptar los cambios. Una vez que esté listo, haces commit y un git push -f

carolosano committed:
Cambios por conflictos en el merge

MVelasquez98 commented:
Ya no tenes más conflictos si te doy aprobación?

MVelasquez98 reviewed (approved):
LGTM

carolosano commented:
Lo que hice fue sacar primero anexo y el resto y hacer push pero todavia o hice lo de develop con el pull lo estoy tratando de hacer ahora profe.

MVelasquez98 commented:
Récorda es MERGE PULL REQUEST cuándo es reléase, está en la consigna. No es rebase and merge, eso es solo con develop. Seguí el instructivo de la consigna

MVelasquez98 commented:
Tenes para escoger en el botón alguna de las dos opciones

MVelasquez98 commented:
![IMG_3520](https://github.com/user-attachments/assets/3a644f31-e237-45d8-a266-1d0ad75c89b1)

carolosano commented:
> ![IMG_3520](https://private-user-images.githubusercontent.com/43156596/470594696-3a644f31-e237-45d8-a266-1d0ad75c89b1.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTM0MDkzMTMsIm5iZiI6MTc1MzQwOTAxMywicGF0aCI6Ii80MzE1NjU5Ni80NzA1OTQ2OTYtM2E2NDRmMzEtZTIzNy00NWQ4LWEyNjYtMWQwYWQ3NWM4OWIxLmpwZWc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNzI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDcyNVQwMjAzMzNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02NTc3M2NkMDllZmJkN2U4OTU5MzU2YTcwZDNjZThlOWY5ODQ3YWM1NDFjMTAwZjI4MDlmYTI3YzU2YjQyY2ExJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.rLBZm-w5orbwtN_3hNXdx7BJ4dFuGThYmy4gPaXOEhc)

aa que raro porque a mi me aparece esto profe por eso
![Uploading Captura de pantalla 2025-07-24 a las 11.05.26 p. m..png…]()

carolosano commented:
> ![IMG_3520](https://private-user-images.githubusercontent.com/43156596/470594696-3a644f31-e237-45d8-a266-1d0ad75c89b1.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTM0MDk1ODAsIm5iZiI6MTc1MzQwOTI4MCwicGF0aCI6Ii80MzE1NjU5Ni80NzA1OTQ2OTYtM2E2NDRmMzEtZTIzNy00NWQ4LWEyNjYtMWQwYWQ3NWM4OWIxLmpwZWc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNzI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDcyNVQwMjA4MDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iMDE0Y2RjYzVhMDhmYzlmNTYxOTk5ZmEzMGEwZjdmMDkyZmMxMjYyOGMwNTgxNTQyMWU1ZDM0MTllM2Q0ZWNmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.aAY1I-HIbgF8RitLT8XRB885eRbgkC-tBtmygOT8BdQ)

aite juro estoy desesperadaporque no entiendonpor que no me dice ese boton !!
